### PR TITLE
replace prefetchq/eslint-config with eslint v9 config

### PR DESCRIPTION
**Pull request by stackoverfloweth**
This PR replaces the outdated @prefecthq/eslint-config with a local eslint.config built on v9

After merging this, I'll pull back out the dependencies/config into something like @kitbag/eslint-config

I also ran `lint:fix` and fixed some manual places

we can ignore the following errors, I'll explicitly override these rules inside this project when I pull out the config

> Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
> Expected an error object to be thrown     @typescript-eslint/only-throw-error

### CODE REVIEW

File: src/types/params.ts [hunk 1]
```diff
@@ -32,11 +32,7 @@ export function isParamGetter(value: Param): value is ParamGetter {
  * @returns True if the value is an object with both 'get' and 'set' functions defined.
  */
 export function isParamGetSet(value: Param): value is ParamGetSet {
-  return typeof value === 'object'
-    && 'get' in value
-    && typeof value.get === 'function'
-    && 'set' in value
-    && typeof value.set === 'function'
+  return typeof value === 'object' && 'get' in value && typeof value.get === 'function' && 'set' in value && typeof value.set === 'function'
```
Comment (stackoverfloweth) [inline src/types/params.ts:35]: flag

File: src/services/hooks.spec.ts [hunk 2]
```diff
@@ -62,10 +62,22 @@ test('calls hook with correct routes', () => {
...
   { type: 'push', status: 'PUSH', hook: (_to, { push }) => push('/') },
   { type: 'replace', status: 'PUSH', hook: (_to, { replace }) => replace('/') },
-  { type: 'abort', status: 'ABORT', hook: (_to, { abort }) => abort() },
+  {
+    type: 'abort',
+    status: 'ABORT',
+    hook: (_to, { abort }) => {
+      abort()
+    },
+  },
```
[the remark this answers is not in the document]
Comment (pleek91) [reply src/services/hooks.spec.ts:80]: Oh I get why it broke it out. The arrow function was returning `abort()` but a separate linting rule says not to return anything when expecting `void`. So it added brackets and whitespace to the arrow function, which caused the object to be multi line, so it broke each property into a line. I'm fine with this.

File: src/components/routerLink.browser.spec.ts [hunk 3]
```diff
@@ -26,7 +26,7 @@ test('renders an anchor tag with the correct href and slot content', () => {
 
   const wrapper = mount(routerLink, {
     props: {
-      to: (resolve) => resolve('parent', { paramName: paramValue }),
+      to: resolve => resolve('parent', { paramName: paramValue }),
```
[the remark this answers is not in the document]
Comment (stackoverfloweth) [reply src/components/routerLink.browser.spec.ts]: updated config for `@stylistic/arrow-parens` from default value to "always"

File: src/types/params.ts (hunk 1 above)
Comment (stackoverfloweth) [reply src/types/params.ts]: this too should be resolved

**Review by pleek91 — APPROVED** [review]

**Pull request merged**